cmd/cvp-trigger/main.go

```diff
@@ -4,69 +4,165 @@ import (
+	fs.StringVar(&o.prowConfigPath, prowConfigPathOption, "", "Path to the Prow config file")
+	fs.StringVar(&o.releaseImageRef, releaseImageRefOption, "", "Pull spec of a specific release payload image used for OCP deployment.")
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Executes a dry-run, displaying the job YAML without submitting the job to Prow")
 }
 
-func gatherOptions() options {
-	o := options{}
-	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+// validateOptions ensures that all required flag options are present and
+// validates any constraints on appropriate values
+func (o options) validateOptions() error {
+	afs := afero.Afero{Fs: fileSystem}
 
-	fs.BoolVar(&o.confirm, "confirm", false, "Whether to actually submit the job to Prow")
-	fs.StringVar(&o.prowConfigPath, prowConfigPathOption, "", "Path to the Prow config file")
-	fs.StringVar(&o.jobConfigPath, jobConfigPathOption, "", "Path to the Prow job config directory")
-	fs.StringVar(&o.periodic, "periodic", "", "Name of the Periodic job to manually trigger")
+	if o.bundleImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", bundleImageRefOption)
+	}
 
-	_ = fs.Parse(os.Args[1:])
-	return o
-}
+	if o.channel == "" {
+		return fmt.Errorf("required parameter %s was not provided", channelOption)
+	}
 
-func (o options) validate() error {
-	if o.prowConfigPath == "" {
-		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	if o.indexImageRef == "" {
+		return fmt.Errorf("required parameter %s was not provided", indexImageRefOption)
 	}
 
 	if o.jobConfigPath == "" {
 		return fmt.Errorf("required parameter %s was not provided", jobConfigPathOption)
 	}
+	exists, _ := afs.Exists(o.jobConfigPath)
+	if !exists {
+		return fmt.Errorf("validating job config path %s failed, does not exist", o.jobConfigPath)
+	}
+
+	if o.jobName == "" {
+		return fmt.Errorf("required parameter %s was not provided", jobNameOption)
+	}
+
+	if o.ocpVersion == "" {
+		return fmt.Errorf("required parameter %s was not provided", ocpVersionOption)
+	}
+	if !strings.HasPrefix(o.ocpVersion, "4") {
+		return fmt.Errorf("ocp-version must be 4.x or higher")
+	}
 
-	if o.periodic == "" {
-		return fmt.Errorf("required parameter %s was not provided", periodicOption)
+	if o.packageName == "" {
+		return fmt.Errorf("required parameter %s was not provided", packageNameOption)
+	}
+
+	if o.prowConfigPath == "" {
+		return fmt.Errorf("required parameter %s was not provided", prowConfigPathOption)
+	}
+	exists, _ = afs.Exists(o.prowConfigPath)
+	if !exists {
+		return fmt.Errorf("validating prow config path %s failed, does not exist", o.prowConfigPath)
 	}
 
 	return nil
 }
 
+// getClusterConfig returns a config from calling util.LoadClusterConfig()
+func getClusterConfig() (*rest.Config, error) {
+	var clusterConfig *rest.Config
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return clusterConfig, nil
```

Comment:
simple `return clusterConfig, err` is equivalent, nothing should rely on the first returned value when err is not `nil` anyway

Reply:
I usually return `nil` just to make sure I don't cheat and that it's painfully obvious that we're in an error situation. I can amend to just do a `return clusterConfig, err` if you would prefer, however.

Reply:
IMO this should just be `util.LoadClusterConfig()` -- the wrapper adds no utility, calling code that attempts to use the config if `err != nil` is malformed